junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -187,13 +209,22 @@ private void assertSupportedType(String target, Class<?> type) {
 		}
 	}
 
-	private Object getPathOrFile(AnnotatedElement sourceElement, Class<?> type, CleanupMode cleanupMode,
-			ExtensionContext extensionContext) {
-		Namespace namespace = getScope(extensionContext) == Scope.PER_DECLARATION //
+	@SuppressWarnings("deprecation")
+	private Object getPathOrFile(AnnotatedElement sourceElement, Class<?> type, TempDirFactory factory,
+			CleanupMode cleanupMode, ExtensionContext extensionContext) {
+		Scope scope = getScope(extensionContext);
+
+		if (scope == Scope.PER_CONTEXT && factory != TempDirFactory.Standard.INSTANCE) {
+			throw new ExtensionConfigurationException("Custom @TempDir factory is not supported with "
```

Comment:
Why is it not supported with that scope?

Reply:
It's basically connected to your comment at https://github.com/junit-team/junit5/issues/2400#issuecomment-1096279042 (I guess).

For example, if `@TempDir` is declared on two fields in the same test class but each annotation has its own factory, I assume the PER_CONTEXT scope would provide the same instance to both, making one of the factories useless. I haven't demonstrated this with a test case yet, but it's something I can try 👍 

If my guess is correct, a local factory configuration should be disallowed due to unpredictable behavior, but a global configuration via the `junit.jupiter.tempdir.factory.default` configuration property (not part of this PR but I'm already working on the changes) could be allowed.

Or do you see it differently?

Reply:
@marcphilipp I was able to confirm my assumption via this example in `TempDirectoryPerContextTests` (without the restriction in `TempDirectory`):

![multiple_temp-dir-factories](https://user-images.githubusercontent.com/26772046/233808967-9f4015b3-c576-405f-a21e-274740c77113.png)

If you check the path injected into `tempDir2` (shown in gray by the debugger), it's the one created by `Factory1` which I wouldn't consider to be the expected behavior.

For this reason, I would keep the restriction. What do you think?

Reply:
Yes, that makes sense. 👍